File: Languages/typescript/src/provider/MinValueAnalysis.ts
import {SimpleTableAnalysis} from '../model/SimpleTableAnalysis'
import { UserInfo } from '../model/UserInfo';

// Classe de analise que implementa a interface "SimpleTableAnalysis"
export class MinValueAnalysis implements SimpleTableAnalysis {

    // Método de interface, responsável por realizar obter o valor minimo de credit em uma lista de usuarios
    // @param userInfoList Lista de usuarios
    // @return Valor minimo de credit
    public analysis(userInfoList: Array<UserInfo>): number{

        var min: number = Number.MAX_VALUE;

        userInfoList.forEach(userInfo => {
            if(min > userInfo.getCredit()) min = userInfo.getCredit()
        });

        return min;
    }
}

File: Languages/python3/src/provider/MinValueAnalysis.py
from src.model.SimpleTableAnalysis import SimpleTableAnalysis
from src.model.UserInfo import UserInfo

# Classe de analise que implementa a interface "SimpleTableAnalysis"
class MinValueAnalysis(SimpleTableAnalysis):

    # Método responsável por obter o menor valor de credit na lista de usuarios
    # @list Lista de usuarios
    # @return Valor minimo da lista
    def analysis(self, userInfoList):
        vmin = 1000000000
        
        for userInfo in userInfoList:
            if(userInfo.getCredit() < vmin): 
                vmin = userInfo.getCredit()

        return vmin
